lib/Language/Haskell/Stylish/Config/Cabal.hs

```diff
@@ -0,0 +1,93 @@
+--------------------------------------------------------------------------------
+module Language.Haskell.Stylish.Config.Cabal
+    ( findLanguageExtensions
+    ) where
+
+
+--------------------------------------------------------------------------------
+import           Data.Either                              (isRight)
+import           Data.List                                (nub)
+import           Data.Maybe                               (maybeToList)
+import qualified Distribution.PackageDescription          as Cabal
+import qualified Distribution.PackageDescription.Parsec   as Cabal
+import qualified Distribution.Simple.Utils                as Cabal
+import qualified Distribution.Types.CondTree              as Cabal
+import qualified Distribution.Verbosity                   as Cabal
+import qualified Language.Haskell.Extension               as Language
+import           Language.Haskell.Stylish.Verbose
+import           System.Directory                         (getCurrentDirectory)
+
+
+--------------------------------------------------------------------------------
+import           Language.Haskell.Stylish.Config.Internal
+
+
+--------------------------------------------------------------------------------
+findLanguageExtensions :: Verbose -> IO [Language.KnownExtension]
+findLanguageExtensions verbose =
+    findCabalFile verbose >>=
+    maybe (pure []) (readDefaultLanguageExtensions verbose)
+
+
+--------------------------------------------------------------------------------
+-- | Find the closest .cabal file, possibly going up the directory structure.
+--   It's essential that
```

Comment:
Oops, sorry, "It's essential that" is probably a part of a thought that didn't fully make it into the docstring from my head. I think it's better to just remove it before merging.